app/src/main/java/com/simplecity/amp_library/ui/screens/album/detail/AlbumDetailFragment.kt

```diff
@@ -420,7 +420,7 @@ class AlbumDetailFragment :
         override fun onSongOverflowClick(position: Int, v: View, song: Song) {
             val popupMenu = PopupMenu(v.context, v)
             SongMenuUtils.setupSongMenu(popupMenu, false, false, playlistMenuHelper)
-            popupMenu.setOnMenuItemClickListener(SongMenuUtils.getSongMenuClickListener(parentFragment as MainController, context!!, song, presenter))
+            popupMenu.setOnMenuItemClickListener(SongMenuUtils.getSongMenuClickListener(null, context!!, song, presenter))
```

Comment:
Can you just pass in whatever `Context` (e.g., Activity) is available here instead of passing in `null`? May avoid potential NPEs if someone assumes that contexts being passed through will always be available.

Reply:
Same for all of these edits

Reply:
@barbeau https://github.com/CUTR-at-USF/MUSER/blob/fix-crash/app/src/main/java/com/simplecity/amp_library/utils/menu/song/SongMenuUtils.kt#L112

I'm checking the condition for null here before passing the context to the callback in the above link. 
Plus the function signature requires the context type `MainController` to be passed and typecasting the available context to it was causing the crash.

Reply:
Ah, right, MainController is a Fragment not an Activity.